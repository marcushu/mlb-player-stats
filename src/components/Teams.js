import React from 'react';
import PropTypes from 'prop-types';

function Teams(props) {

  return (
    <>
        {props.teamsArray.map(team =>
          <span key={team.team_id}>
            <img
              src={require(`../images/${team.gif}`)}
              style={{ width: "60px" }}
              alt="test"
              onClick={() => props.callBack(team.team)}
            />
          </span>
        )}
      </>
  )
}

Teams.propTypes = {
  callBack: PropTypes.func.isRequired,
  teamsArray: PropTypes.array.isRequired
}

export default Teams;