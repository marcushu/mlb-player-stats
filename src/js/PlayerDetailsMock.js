const hitting = {
    "created":"2020-08-12T21:51:33",
    "totalSize":"1",
    "row":
    {"hr":"165","gidp":"67","np":"13519","sac":"0","team_count":"4","sport_code":"mlb","hgnd":"272","rbi":"528","lob":"1331","babip":".305","gidp_opp":"544","tb":"1586","xbh":"362","sport_id":"1","bb":"240","avg":".273","slg":".497","ops":".824","hbp":"31","g":"834","d":"173","hfly":"168","so":"742","wo":"3","sport":"MLB","sf":"28","hpop":"6","tpa":"3490","h":"870","cs":"20","obp":".327","hldr":"424","t":"24","ao":"939","r":"475","go_ao":"0.78","sb":"43","ppa":"3.87","player_id":"493316","ab":"3191","ibb":"33","roe":"37","go":"734"}}

const seasonHitting = {
    "created":"2020-08-12T21:51:33",
    "totalSize":"1",
    "row": {}
}    

const pitching = {
    "created":"2020-08-12T21:51:33",
    "totalSize":"1",
    "row": {}
}

const seasonPitching = {
    "created":"2020-08-12T21:51:33",
    "totalSize":"1",
    "row": {}
}

export { hitting, seasonHitting, pitching, seasonPitching }