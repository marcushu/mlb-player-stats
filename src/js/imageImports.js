
import bravesImg from '../images/braves.gif';
import indiansImg from '../images/indians.gif';
import oriolesImg from '../images/orioles.gif';
import redSoxImg from '../images/redSox.gif';
import diamodBacks from '../images/diamondBacks.gif';
import sox from '../images/sox.gif';
import reds from '../images/reds.gif';
import cubs from '../images/cubs.gif';
import angels from '../images/angels.gif';
import as from '../images/as.gif';
import astros from '../images/astros.gif';
import blueJays from '../images/blueJays.gif';
import cardinals from '../images/cardinals.gif';
import dodgers from '../images/dodgers.gif';
import mariners from '../images/mariners.gif';
import marlins from '../images/marlins.gif';
import mets from '../images/mets.gif';
import padres from '../images/padres.gif';
import phillies from '../images/phillies.gif';
import pirates from '../images/pirates.gif';
import rangers from '../images/rangers.gif';
import rays from '../images/rays.gif';
import royals from '../images/royals.gif';
import yankees from '../images/yankees.gif';
import nationals from '../images/nationals.gif';
import giants from '../images/giants.gif';
import tigers from '../images/tigers.gif';
import brewers from '../images/brewers.gif'
import twins from '../images/twins.gif';
import rockies from '../images/rockies.gif';

export {
    rockies,
    nationals,
    giants,
    tigers,
    brewers,
    twins,
    bravesImg,
    indiansImg,
    oriolesImg,
    redSoxImg,
    diamodBacks,
    sox,
    reds,
    cubs,
    angels,
    as,
    astros,
    blueJays,
    cardinals,
    dodgers,
    mariners,
    marlins,
    mets,
    padres,
    phillies,
    pirates,
    rangers,
    rays,
    royals,
    yankees,
}


