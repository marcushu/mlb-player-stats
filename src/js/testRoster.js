const tstRoster = [
    {
      "position_txt": "1B",
      "weight": "245",
      "name_display_first_last": "Pete Alonso",
      "college": "Florida",
      "height_inches": "3",
      "starter_sw": "N",
      "jersey_number": "20",
      "end_date": "",
      "name_first": "Peter",
      "bats": "R",
      "team_code": "nyn",
      "height_feet": "6",
      "pro_debut_date": "2019-03-28T00:00:00",
      "status_code": "A",
      "primary_position": "3",
      "birth_date": "1994-12-07T00:00:00",
      "team_abbrev": "NYM",
      "throws": "R",
      "team_name": "New York Mets",
      "name_display_last_first": "Alonso, Pete",
      "name_use": "Pete",
      "player_id": "624413",
      "name_last": "Alonso",
      "team_id": "121",
      "start_date": "2019-03-28T00:00:00",
      "name_full": "Alonso, Pete"
    },
    {
      "position_txt": "P",
      "weight": "265",
      "name_display_first_last": "Dellin Betances",
      "college": "",
      "height_inches": "8",
      "starter_sw": "N",
      "jersey_number": "68",
      "end_date": "",
      "name_first": "Dellin",
      "bats": "R",
      "team_code": "nyn",
      "height_feet": "6",
      "pro_debut_date": "2011-09-22T00:00:00",
      "status_code": "A",
      "primary_position": "1",
      "birth_date": "1988-03-23T00:00:00",
      "team_abbrev": "NYM",
      "throws": "R",
      "team_name": "New York Mets",
      "name_display_last_first": "Betances, Dellin",
      "name_use": "Dellin",
      "player_id": "476454",
      "name_last": "Betances",
      "team_id": "121",
      "start_date": "2019-12-24T00:00:00",
      "name_full": "Betances, Dellin"
    }
]

export default tstRoster