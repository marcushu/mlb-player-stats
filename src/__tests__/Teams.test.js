import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Teams from '../components/Teams';
import { americanLeague } from '../js/teamIds';

describe("Teams component", () => {

    it('renders', () => {
        
        const { asFragment } = render(
            <Teams callBack={() => { }} teamsArray={americanLeague} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('calls prop/function on click', () => {
        const mockFunc = jest.fn();

        const { getAllByAltText } = render(
            <Teams callBack={mockFunc} teamsArray={americanLeague} />
        );
        const nodes = getAllByAltText("test");

        fireEvent.click(nodes[0]);

        expect(mockFunc).toHaveBeenCalled();
    });
});
