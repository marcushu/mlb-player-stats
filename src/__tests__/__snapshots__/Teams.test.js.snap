// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teams component renders 1`] = `
<DocumentFragment>
  <span>
    <img
      alt="test"
      src="orioles.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="redSox.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="sox.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="indians.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="tigers.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="astros.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="royals.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="angels.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="twins.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="yankees.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="as.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="mariners.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="rays.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="rangers.gif"
      style="width: 60px;"
    />
  </span>
  <span>
    <img
      alt="test"
      src="blueJays.gif"
      style="width: 60px;"
    />
  </span>
</DocumentFragment>
`;
